cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/SystemDatasetRuntimeModule.java

```diff
@@ -88,7 +117,7 @@ protected void configure() {
                                                                             Names.named("defaultDatasetModules"));
 
         // NOTE: order is important due to dependencies between modules
-        mapBinder.addBinding("orderedTable-hbase").toInstance(new HBaseTableModule());
+        mapBinder.addBinding("orderedTable-hbase").toProvider(OrderedTableModuleProvider.class);
```

Comment:
I think this need to be a singleton binding (before it was binded with toInstance, which is effectively a singleton). Simply add '.in(Scopes.SINGLETON)` will do the job.